src/runtime/propensity.js

```diff
@@ -69,18 +75,35 @@ export class Propensity {
 
   /** @override */
   sendEvent(userEvent) {
-    if (!Object.values(PropensityApi.Event).includes(userEvent.name)) {
-      throw new Error('Invalid user event provided');
-    }
-    if (userEvent.data && !isObject(userEvent.data)) {
-      throw new Error('Event data must be an Object');
+    const analyticsEvent = propensityEventToAnalyticsEvent(userEvent.name);
+    if (!isEnumValue(PropensityApi.Event, userEvent.name)
+        || !analyticsEvent) {
+      throw new Error('Invalid user event provided(' + userEvent.name + ')');
     }
-    // TODO(sohanirao, mborof): Idenfity the new interface with event
-    // manager and update the lines below to adhere to that interface
-    let paramString = null;
-    if (userEvent.data) {
-      paramString = JSON.stringify(userEvent.data);
+
+    //event manager requires data to be a real object so fill it in if
+    //the publisher didn't pass the field in
+    if (!userEvent.data) {
```

Comment:
Line 86, 87 feels like redundant code

Reply:
if (userEvent.data && !isObject(userEvent.data)) {
      throw new Error('Event data must be an Object');
    }

Any reason why this code was not sufficient?

Reply:
It caused the test code to fail so I figured it was supposed to work

Reply:
It didn't fail before because nothing had a hard requirement that this was an object until event manager

Reply:
Here's a piece of code that tested it already
it('should provide valid event', () => {
    const correctEvent = {
      name: PropensityApi.Event.IMPRESSION_PAYWALL,
      active: false,
    };
    expect(() => {
      propensity.sendEvent(correctEvent);
    }).to.not.throw('Invalid user event provided');
    const incorrectEvent = {
      name: 'user-redirect',
    };
    expect(() => {
      propensity.sendEvent(incorrectEvent);
    }).to.throw('Invalid user event provided');
    const incorrectEventParam = {
      name: PropensityApi.Event.IMPRESSION_OFFERS,
      active: true,
      data: 'all_offers',
    };
    expect(() => {
      propensity.sendEvent(incorrectEventParam);
    }).to.throw(/Event data must be an Object/);
  });

Still not sure how your code is different other than splitting up the checks into two lines?

Reply:
Here is the specific test that used to work that would no longer work if I throw an error without a data object:

    const correctEvent = {
      name: PropensityApi.Event.IMPRESSION_PAYWALL,
      active: false,
    };
    expect(() => {
      propensity.sendEvent(correctEvent);
    }).to.not.throw('Invalid user event provided');